src/contexts/Api.tsx

```diff
@@ -1,71 +1,52 @@
 import React, { useContext, createContext, useEffect, useState } from 'react';
 import { ApiPromise } from '@polkadot/api';
-import { NotificationType, useNotifications } from './Notification';
-import useSearchParam from '../hooks/useSearchParam';
+import { useEndpointsParams } from '../hooks';
 import { measured } from '../utils/performance';
-import {
-  endpointFor,
-  Network,
-  networkFor,
-  newApi,
-} from '../utils/polkadot-api';
+import { newApi } from '../utils/polkadot-api';
 import { timeout } from '../utils/promise';
+import { NotificationType, useNotifications } from './Notification';
 
 const CHAIN_CONNECTION_TIMEOUT = 5000; // in milliseconds
 
-export interface IApiContext {
-  api: ApiPromise | undefined;
-}
+const apiContext = createContext({} as ApiPromise | undefined);
 
-// api context
-const apiContext = createContext({} as IApiContext);
 export const useApi = () => useContext(apiContext);
 
 const ApiProvider = ({ children }: { children: React.ReactNode }) => {
-  const [api, setApi] = useState<ApiPromise>();
-  const networkParam = useSearchParam('network');
-  const network = Network.parse(networkParam);
-  const rpcParam = useSearchParam('rpc');
   const { notify } = useNotifications();
+  const endpoints = useEndpointsParams();
+  const [api, setApi] = useState<ApiPromise>();
 
   useEffect(() => {
     async function fetch() {
-      const endpoint = rpcParam ? rpcParam : endpointFor(network);
-      try {
-        const api = await measured('api', () =>
-          timeout(newApi(endpoint), CHAIN_CONNECTION_TIMEOUT)
-        );
-        if (rpcParam) {
-          // Check that provided rpc and network point to a same logical chain
-          if (networkFor(api) != network) {
-            const message = `Provided RPC doesn't match network ${network}: ${rpcParam}`;
-            const notification = { type: NotificationType.Error, message };
-            notify(notification);
-          } else {
-            const message = `Connected to network ${network} using RPC ${rpcParam}`;
-            const notification = {
-              type: NotificationType.Notification,
-              message,
-            };
-            console.info(message);
-            notify(notification);
-          }
-        } else {
-          const message = `Connected to network ${network.toString()}`;
-          const notification = { type: NotificationType.Notification, message };
-          console.info(message);
-          notify(notification);
+      if (endpoints.type == 'ok') {
+        try {
+          const api = await measured('api', () =>
+            timeout(newApi(endpoints.value), CHAIN_CONNECTION_TIMEOUT)
+          );
```

Comment:
Is this context necassary now that the api is avilable in the reducer state. I believe the api from the reducer can be provider with a context provider as a context with a useApi (The reducer that is in the App.tsx)